### PR TITLE
Markdown: Ensure isomorphism at serialization/deserialization

### PR DESCRIPTION
**Pull request by jpreynat**
The issue that this PR solves is that serializing then deserializing custom blocks in markdown when they end with complex blocks (such as a list or a blockquote) is not isomorphic.

Indeed, while deserializing custom blocks, some of these blocks need to capture the next line.
Since at serialization, newlines are trimmed, custom blocks ending tag are captured, which lead the parser to put them in the wrong place. They are finally removed during the normalization process of Slate, since the document is now invalid.

For instance, a blockquote at the end of a custom block:
```
{% someblock %}
Here is some text.

> And a quote.
{% endsomeblock %}
```
will lead to the blockquote deserialization regex to capture the last 2 lines `And a quote.\n{% endsomeblock %}`.

To fix this issue, the serialization to markdown now always adds a newline after the start tag of a custom block and before the ending tag of the block, if any.

So basically, instead of converting some documentation to the markdown above, it will lead to:
```
{% someblock %}

Here is some text.

> And a quote.

{% endsomeblock %}
```
which is then parsed correctly.

This is a potential "breaking" change, since all custom blocks will now be serialized this way into markdown.
If this PR is merged, it would thus be a good thing to bump to a major version.

### CODE REVIEW

**Comment by jpreynat** [comment]
Closed in favor of #139
